### DISCUSSION TITLE
A mistake in the documentation?

### DISCUSSION THREAD

**Discussion by AshMosh** [pytest-dev/pytest#11272] [General]
Looking into the documentation regarding fixtures, I identified two possible mistakes:
1. [Here](https://docs.pytest.org/en/6.2.x/fixture.html#what-fixtures-are:~:text=modify%20state%20is-,neglibible,-%2C%20as%20most%20of) maybe **negligible**?
2. [Here](https://docs.pytest.org/en/6.2.x/fixture.html#what-fixtures-are:~:text=If%20driver%20executes%20before%20user%2C%20and%20user%20raises%20an%20exception%2C%20the%20driver%20will%20still%20quit%2C%20and%20the%20user%20was%20never%20made.%20And%20if%20driver%20was%20the%20one%20to%20raise%20the%20exception%2C%20then%20the%20driver%20would%20never%20have%20been%20started%20and%20the%20user%20would%20never%20have%20been%20made.) I think the right phrasing should be:
" If driver executes before user, and user raises an exception, the driver will still quit, and the user was never made. And if driver was the one to raise the exception, **then the driver would never have been made and the user would never have been started**.

Not sure if there a better place to raise these issues, let me know if there is.

**Comment by nicoddemus** [comment]
Hey @AshMosh, thanks for raising these.

 > Not sure if there a better place to raise these issues, let me know if there is.

You can open a PR directly, and even if not correct at first, we are happy to guide you there. So opening a PR with your suggestions would be good step. 😁

**Comment by The-Compiler** [comment]
Also note that you're looking at the 6.2.x docs - the last release from that branch was around two years ago, and e.g. the first typo you pointed out was long fixed already (in 100c8deab5fccda043b2087bd450d074c1f8756d).

For the second one, I think the current wording seems correct? You start a driver and create a user - I don't think creating a driver and starting a user makes sense.

**Reply by AshMosh** [reply]
The original quote was:
>  If driver executes before user, and user raises an exception, the driver will still quit, and the user was never made. And if driver was the one to raise the exception, then the driver would never have been started and the user would never have been made.

By the first part you see that because user failed, it was never made (we assume that for failures), thus by the same logic, if driver failed, it was never made, but because it was executed before user, we don't get to the point that user even starts, therefore my suggestion makes more sense.

**Reply by The-Compiler** [reply]
I still don't follow. The relevant code is:

```python
    # creates an user
    _user = User(name="Susan", username=f"testuser-{uuid4()}", password="P4$$word")
    admin_client.create_user(_user)
```

and

```python
    # starts a driver
    _driver = Chrome()
```

thus, if `User()` fails, the user is not made/created, and if `Chrome()` fails, the driver is not started.

**Reply by AshMosh** [reply]
Let's compare the objects that raise the exception:
user raises an exception -> user was never made
driver raises an exception -> driver would never have been started
why don't they both get the same result?

**Reply by The-Compiler** [reply]
But it is the same result... just with a verb that makes more sense in the context of the code shown. "Creating a user" and "starting a driver" isn't something pytest does, it's a description of what the quoted code does.
